File: backend/app/schemas/analytics.py
from pydantic import BaseModel
from typing import List, Optional

class AnalyticsSummary(BaseModel):
    totalEvents: int
    criticalEvents: int
    highRiskEvents: int
    mediumRiskEvents: int
    lowRiskEvents: int
    preventionIndex: float = 78.0
    activeIncidents: int = 0
    resolvedIncidents: int = 0
    repeatBehaviourPct: float = 18.0

class AnalyticsSummaryResponse(BaseModel):
    summary: AnalyticsSummary

class BehaviourMetric(BaseModel):
    name: str
    value: int
    avg_score: Optional[float] = None

class RiskMetric(BaseModel):
    risk_level: str
    count: int
    avg_score: Optional[float] = None

class TimelinePoint(BaseModel):
    timestamp: float
    event_id: str
    behaviour: str
    risk_score: float
    risk_level: str
    bay_id: Optional[str] = None

File: frontend/src/types/analytics.ts
export interface AnalyticsSummary {
  totalEvents: number;
  criticalEvents: number;
  highRiskEvents: number;
  mediumRiskEvents: number;
  lowRiskEvents: number;
  preventionIndex?: number;
  activeIncidents?: number;
  resolvedIncidents?: number;
  repeatBehaviourPct?: number;
}

export interface AnalyticsSummaryResponse {
  summary: AnalyticsSummary;
}

export interface AnalyticsData {
  summary: AnalyticsSummary;
  behaviours: BehaviourMetric[];
}

export interface BehaviourMetric {
  name: string;
  value: number;
  avg_score?: number;
}

export interface RiskMetric {
  risk_level: string;
  count: number;
  avg_score?: number;
}

export interface TimelinePoint {
  timestamp: number;
  event_id: string;
  behaviour: string;
  risk_score: number;
  risk_level: string;
  bay_id?: string;
}
